### PR TITLE
2 - Jf rawdataprocess fmt

### PR DESCRIPTION
**Pull request by vhinger182**
Implement filename parsing using fmt::format + bug fixes

### CODE REVIEW

**Review by erikfrojdh — CHANGES_REQUESTED** [review]
Could you fix the const&? If the functions are to be used in other files we could break out the functions and add also a small test... but that's for next time.

File: slsDetectorCalibration/jungfrauExecutables/jungfrauRawDataProcess.cpp [hunk 1]
```diff
@@ -41,12 +41,38 @@
 #include <ctime>
 #include <fmt/core.h>
 
+std::string getRootString( const std::string filepath ) {
```
[the remark this answers is not in the document]
Comment (vhinger182) [reply slsDetectorCalibration/jungfrauExecutables/jungfrauRawDataProcess.cpp:44]: True! I'll change it.
Just to be sure about the workflow: Do I simply make a new commit with the changes on my local branch and push it? Or can I do it differently?

[the remark this answers is not in the document]
Comment (erikfrojdh) [reply slsDetectorCalibration/jungfrauExecutables/jungfrauRawDataProcess.cpp:44]: Easiest is to make a commit to jf_rawdataprocess_fmt on your machine and push it to github. That will automatically update the PR

File: slsDetectorCalibration/jungfrauExecutables/jungfrauRawDataProcess.cpp [hunk 2]
```diff
@@ -41,12 +41,38 @@
...
+
+//Create file name string according to slsDetectorPackage format
+//   dir:     directory
+//   fprefix:   fileprefix (without extension)
+//   fsuffix:   filesuffix (for output files, e.g. "ped")
+//   fext:    file extension (e.g. "raw")
+//   mindex:  module index ("d0" in standard)
+//   findex:  file index for one acquisition ("f0")
+//   aindex:  acquisition index (i.e. "run number")
+std::string createFileName( const std::string dir, std::string fprefix="run", std::string fsuffix="", std::string fext="raw", int aindex=0, int mindex=0, int findex=0, int outfilecounter=-1 ) {
```
[the remark this answers is not in the document]
Comment (vhinger182) [reply slsDetectorCalibration/jungfrauExecutables/jungfrauRawDataProcess.cpp:59]: The ones that have a default value I cannot make const& right?

[the remark this answers is not in the document]
Comment (vhinger182) [reply slsDetectorCalibration/jungfrauExecutables/jungfrauRawDataProcess.cpp]: Ok, fixed it! :)

**Review by erikfrojdh — APPROVED** [review]

**Pull request merged**